arrangor/src/main/kotlin/no/nav/amt/tiltak/veileder/ArrangorVeilederRepository.kt

```diff
@@ -150,4 +171,34 @@ open class ArrangorVeilederRepository(
 		val parameters = sqlParameters("deltakerIder" to deltakerIder)
 		return template.query(sql, parameters, rowMapper)
 	}
+
+	internal fun getDeltakerlisteForVeileder(ansattId: UUID): List<ArrangorVeiledersDeltaker> {
```

Comment:
Hva tenker dere om denne? Bør dette gå fint, eller blir den for tung?

Reply:
Tung blir den nok uansett 😅 Vi hadde en diskusjon rundt repository patternet, og hvor vi da var til dels enige om å unngå joins her, da det krysser litt på tvers av domenet, og repositoryet burde være forbeholdt ett domeneobjekt. Man må selfølgelig være litt pragmatisk her, og det er ikke noe vi følger til punkt og prikke som du sikkert har sett i koden vår.

Måten deltakerlisten sys sammen på for koordinator siden kan du se i `ControllerService`, det er ikke pent det heller, men det var den veien vi gikk for en stund siden. Før det var det QueryObjekter for slike store joins. 

Jeg vet ikke hva som er best. Det å bruke joins slik som dette er sikkert mer performant enn å gjøre flere spørringer, men det klusser til domenet nokså mye, og er egentlig kun for å serve front-end.

Reply:
Det koster generelt veldig mye mer å gjøre flere spørringer enn å gjøre en join :/ Det fører fort til veldig trege kall for sluttbruker, og man plager databasen en god del mer enn nødvendlig. Hvis man har tatt et aktivt valg om å ikke bruke joins i spørringer så kan det hende vi burde gå bort fra å ha informasjon spredd i så mange tabeller..

Reply:
Vær så snill ikke begynn med query pattern. Det hadde vi en diskusjon på og endte opp med å fjerne. Denne pr'en er jo et godt eksempel på hvor mye enklere det er når man bruker joins og lar databasen håndtere det den er god på isteden for å la repository pattern styre fremgangsmåte. Er sikker på at det er mye mer effektivt og det er jo noe vi begynner å kjenne på nå, at ytelsen i appen vår er ikke god nok. 

Fordelen med å delegere til de forskjellige servicene/repositoryene ligger i at man lettere kan splitte opp servicene i nye apper senere, uten å endre på for mye. Men det forutsetter jo uansett at den oppdelingen vi har valgt er den som skal videreføres. 

Sånn sett så syns jeg det er litt vanskelig å ta stilling til det her. Hvis vi skal komme igang med å dele opp appen ganske fort så ville jeg splitta opp spørringen, men hvis den skal leve lenge så tror jeg det ville ført til at vi bare lager noe som må fikses

Reply:
Jeg har ikke noe imot å bruke joins, jeg bare synes det er viktig at vi alle er enige om vi går for å endre fremgangsmåten vår på. Vi kan gjerne teste dette sånn, jeg er også litt nysgjerrig på hvor stor forskjell det er mellom de to måtene å gjøre det på, kallet på å hente deltakerlisten for koordinator ligger på noen hundre ms så hvis dette får det drastisk ned så er jo det bra.